### ISSUE TITLE
Options filter is not cleared properly

### ISSUE THREAD

**Issue by ramkumarc** [jsumners/node-activedirectory#26]
In the below given code, findUser function always gives the user1 details. Same code works fine with the [activedirectory](https://www.npmjs.com/package/activedirectory). The workaround that I am doing right now is to clone the options object and pass it new every time.

```
var activeDirectory = require('activedirectory2');

var ad = new activeDirectory({
    url: '<LDAP URL>',
    baseDN: '<BASE DN>',
    username: '<USER NAME>',
    password: '<PASSWORD'
});

var options = {
    attributes: ["givenName", "sn", "cn", "title", "employeeType"]
};

function findUser(userId) {

    ad.findUser(options, userId, function (err, user) {

        try {
            if (err) {
                console.error('Active Directory Error : ' + JSON.stringify(err));
            }
            else if (!user) {
                console.error('User ' + userId + ' not found in AD');
            }
            else {
                console.log(user);
            }
        } catch (e) {
            console.error(e);
        }
    });
}

findUser("user1");
findUser("user2");
findUser("user3");
```

**Comment by jsumners** [comment]
Thank you for reporting the issue. It is fixed in v1.2.5.

**Comment by ramkumarc** [comment]
Thank you for fixing it quickly.
